src/lib/useMessageCache.ts

```diff
@@ -0,0 +1,81 @@
+import { StreamingStepInput } from "./types";
+
+function useMessageCache(
+  setConversationId: React.Dispatch<React.SetStateAction<number>>,
```

Comment:
Don't think this is _important_, but my pycharm linter seems to think we should be importing `React` in this file to use it :shrug:

Reply:
okay mine didn't say anything, but I'll add it then